docs/ReleaseNotes.md

```diff
@@ -1,6 +1,58 @@
+ - The [PX1062](diagnostics/PX1062.md) diagnostic has been added to check that graphs and graph extensions do not contain statis members.
+ - The following new diagnostics have been added to prevent serealization and desealization errors: 
+   - The [PX1063](diagnostics/PX1063.md) diagnostic checks that the exception-derived classes always declare a serealization constructor.
+   - The [PX1064](diagnostics/PX1064.md) diagnostic checks that the exception-derived classes which declare new serealizable data, always provide an override of the `GetObjectData` method.
+ - Support for the following complex expressions passed to SetProcessDelegate API has been added to the [PX1008](diagnostics/PX1008.md) and [PX1086](diagnostics/PX1086.md) diagnostics:
+   - Cast expression like
+   - Object creation expression
+   - Local variable or parameter that store delegate
+   - Property or field that store delegate
+ - The [PX1056](diagnostics/PX1056.md) diagnostic has been added to check and warn if a graph is created in the `IsActive` method in graph and DAC extension and the `IsActiveForGraph` method.
+ - The [PX1057](diagnostics/PX1057.md) and [PX1084](diagnostics/PX1084.md) diagnostics now prevent graph creation via a direct constructor call, that is, the `new PXGraph()` method call.
+ - The [PX1050](diagnostics/PX1050.md) diagnostic now displays a warning for interpolated strings and strings passed to localization methods or exception constructors that are not constant fields.
+ 
+ #### Code Snippets
+ Acuminator now provides code snippets for DACs, DAC fields, and graph events. A developer can use these code snippets to speed up writing new code and avoid common mistakes.
+   For more details on installing and using the code snippets, see [the CodeSnippets repository on GitHub](https://github.com/Acumatica/CodeSnippets). 
+   
+ #### Improvements in the PX1008 diagnostic 
+ The [PX1008](diagnostics/PX1008.md) diagnostic has been completely reworked. Now it is a data flow diagnostic that goes recursively into methods and checks how the data is transferred to them.
+ The following enhancements have been implemented in the diagnostic:
+   - Check for captured adapters has been implemented 
+   - Improvements in diagnostic messages
+   - Support for the PX1008 diagnostic has been added to custom views and attributes 
+   - Extended design of syntax walkers for inter-procedural analysis
+     New extension points and new API has been added.
+ 
+ #### Improvements in the PX1072 diagnostic 
+ The following fixes have been implemented in the [PX1072](diagnostics/PX1072.md) diagnostic:
+   - In static graph members, the declaring graph now is not considered available for usage in fBQL queries
+   - The diagnostic now does not display a warning if a single local graph is available and used in an fQBL query.
+   - The diagnostic now displayed a warning when a class contains multiple fBQL queries and no graph is available.
```

Comment:
Here it is a little bit more tricky than that. The diagnostic looks for _available_ graph in the context. The available means already existing graphs. This could be:
- `this` if we are inside graph or graph extension
- method parameter of a graph type
- local variable of a graph type.
- In theory it also could be fields and properties but we explicitly do not analyze them and exclude them from the list of available graphs. **This is also a change introduced in this release and it should be mentioned in release notes**

The case described in this line is when you have no such available graphs and multiple BQL queries. Each query uses its own graph by creating it:
```csharp
var queryRes1 = PXSelect<Dac>.Select(new PXGraph());
var queryRes2 = PXselect<AnotherDac>.Select(PXGraph.CreateInstance<SomeGraph>());
```
In this case you can introduce a shared local variable with a graph which can be reused by both queries and now the diagnostic suggests this. I think we should add these details to the release notes.